src/apify_client/clients/resource_clients/key_value_store.py

```diff
@@ -0,0 +1,152 @@
+import io
+import json
+from typing import Any, Dict, Optional
+
+from ..._errors import ApifyApiError
+from ..._utils import _catch_not_found_or_throw, _parse_date_fields, _pluck_data
+from ..base.resource_client import ResourceClient
+
+
+class KeyValueStoreClient(ResourceClient):
+    """Sub-client for manipulating a single key-value store."""
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initializes the KeyValueStoreClient."""
+        super().__init__(*args, resource_path='key-value-stores', **kwargs)
+
+    def get(self) -> Optional[Dict]:
+        """Retrieves the key-value store.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/store-object/get-store
+
+        Returns:
+            The retrieved key-value store
+        """
+        return self._get()
+
+    def update(self, new_fields: Dict) -> Optional[Dict]:
+        """Updates the key-value store with specified fields.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/store-object/update-store
+
+        Args:
+            new_fields (dict): The fields of the key-value store to update
+
+        Returns:
+            The updated key-value store
+        """
+        return self._update(new_fields)
+
+    def delete(self) -> None:
+        """Deletes the key-value store.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/store-object/delete-store
+        """
+        return self._delete()
+
+    def list_keys(self, *, limit: int = None, exclusive_start_key: str = None) -> Any:
+        """Lists the keys in the key-value store.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/key-collection/get-list-of-keys
+
+        Args:
+            limit (int, optional): Number of keys to be returned. Maximum value is 1000
+            exclusive_start_key (str, optional): All keys up to this one (including) are skipped from the result
+        """
+        request_params = self._params(
+            limit=limit,
+            exclusiveStartKey=exclusive_start_key,
+        )
+
+        response = self.http_client.call(
+            url=self._url('keys'),
+            method='GET',
+            params=request_params,
+        )
+
+        return _parse_date_fields(_pluck_data(response.json()))
+
+    def get_record(self, key: str, *, as_bytes: bool = False, as_file: bool = False) -> Optional[Dict]:
```

Comment:
what about creating `Enum` for type of response instead of two boolean flags? I think would be cleaner that way

Reply:
Yeah, it's not ideal, we want to update the whole get_record logic in both the JS and Python clients and maybe split it into multiple functions. I'd go for `get_record`, `download_record` and `stream_record`, but the JS client has to have it somehow backwards compatible, so we still need to figure this out. I'll leave it like it is for now and revisit it later.

Reply:
ok, please add TODO